README.md

```diff
@@ -527,42 +530,291 @@ An example that uses the `ContextualTargetingFilter` in a console application is
+
+### Consumption
+
+Dynamic features are accessible through the `IDynamicFeatureManager` interface.
+
+``` C#
+public interface IDynamicFeatureManager
+{
+    IAsyncEnumerable<string> GetDynamicFeatureNamesAsync(CancellationToken cancellationToken = default);
+
+    ValueTask<T> GetVariantAsync<T>(string dynamicFeature, CancellationToken cancellationToken = default);
+
+    ValueTask<T> GetVariantAsync<T, TContext>(string dynamicFeature, TContext context, CancellationToken cancellationToken = default);
+}
+```
+
+The dynamic feature manager performs a resolution process that takes the name of a feature and returns a strongly typed value to represent the variant's value.
+
+The following steps are performed during the retrieval of a dynamic feature's variant
+1. Lookup the configuration of the specified dynamic feature to find the registered variants
+2. Assign one of the registered variants to be used.
+3. Resolve typed value based off of the assigned variant.
+
+The dynamic feature manager is made available by using the `AddFeatureManagement` call. Make sure to add any required feature variant assigners referenced by dynamic features within the application by using `AddFeatureVariantAssigner`.
+
+``` C#
+using Microsoft.FeatureManagement;
+using Microsoft.FeatureManagement.Assigners;
+
+public class Startup
+{
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services.AddFeatureManagement()
+                .AddFeatureVariantAssigner<TargetingFeatureVariantAssigner>();
+    }
+}
+```
+
+### Usage Example
+
+One possible example of when variants may be used is in a web application when there is a desire to test different visuals. In the following examples a mock of how one might assign different variants of a web page background to their users is shown.
+
+``` C#
+//
+// Modify view based off multiple possible variants
+model.BackgroundUrl = featureVariantManager.GetVariantAsync<string>("HomeBackground", cancellationToken);
+
+return View(model);
+```
+
+### Dynamic Feature Declaration
+
+Dynamic features can be configured in a configuration file similarly to feature flags. Instead of being defined in the `FeatureManagement:FeatureFlags` section, they are defined in the `FeatureManagement:DynamicFeatures` section. Additionally, dynamic features have the following properties.
+
+* Assigner: The assigner that should be used to select which variant should be used any time this feature is accessed.
+* Variants: The different variants of the dynamic feature.
+  * Default: Whether the variant should be used if no variant could be explicitly assigned. One and only one default variant is required.
+  * Configuration Reference: A reference to the configuration of the variant to be used as typed options in the application.

```

Comment:
it should "ConfigurationReference" (no space between "Configuration" and "Reference"). Same for "AssignmentParameters" below.